networking/src/p2p/binary_message.rs

```diff
@@ -161,4 +160,60 @@ impl<T: BinaryMessage> MessageHash for T {
         let bytes = self.as_bytes()?;
         Ok(blake2b::digest(&bytes))
     }
+}
+
+/// Message hash error
+#[derive(Debug, Fail)]
+pub enum MessageHexableError {
+    #[fail(display = "Message to hex serialization error")]
+    ToHexSerializationError {
+        error: ser::Error
+    },
+    #[fail(display = "Message from hex serialization error")]
+    FromHexSerializationError {
+        error: hex::FromHexError
+    },
+    #[fail(display = "Message from binary serialization error")]
+    FromBinarySerializationError {
+        error: BinaryReaderError
+    },
+}
+
+impl From<ser::Error> for MessageHexableError {
+    fn from(error: ser::Error) -> Self {
+        MessageHexableError::ToHexSerializationError { error }
+    }
+}
+
+impl From<hex::FromHexError> for MessageHexableError {
+    fn from(error: hex::FromHexError) -> Self {
+        MessageHexableError::FromHexSerializationError { error }
+    }
+}
+
+/// Trait for converting messages from/to HEX string
+pub trait Hexable: Sized {
```

Comment:
@mrjoe7 
pondelok to prerobim na bytes aby sa posielali hore dole,
potom cele toto Hexable nebude ani treba a dost sa to zjednodusi